File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int zeros = 0, first = 0, count = 0;
        
        for(auto last = 0; last < nums.size(); last++) {
            if(nums[last] == 0)
                zeros++;
            
            while(zeros > 1) {
                if (nums[first] == 0)
                    zeros--;
                first++; 
            }
            count = max(count, last - first);
        }
        return count;
    }
};

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.go
func longestSubarray(nums []int) int {
    l, r := 0, 0
    maxL := 0
    deleted := 0 

    for r < len(nums) {
        if nums[r] != 1 {
            deleted++
        }

        for deleted > 1 {
            if nums[l] != 1 {
                deleted--
            }
            l++
        }

        maxL = max(maxL, r - l)
        r++
    }

    return maxL
}